### DISCUSSION TITLE
Regarding ww3_prnc, ww3_ounf, ww3_bounc, ww3_ounp, and ww3_prtide

### DISCUSSION THREAD

**Discussion by krishnanathira** [NOAA-EMC/WW3#765] [Q&A] [answered]
Hi 
I have installed WW3 6.07.01 through **w3_automake** and now the executables **ww3_prnc, ww3_ounf, ww3_bounc, ww3_ounp, and ww3_prtide** are not present in the **exe** folder. Instead **ww3_grid ww3_strt ww3_prep ww3_shel ww3_ounf ww3_prnc ww3_bounc ww3_ounp ww3_prtide** all these .inp files are present in **inp** folder. I tried to create them by w3_make ww3_prnc, ww3_ounf, ww3_bounc, ww3_ounp, and ww3_prtide. But shows error as follows

Processing ww3_prnc
---------------------
ad3 : processing ww3_prnc 
mv: cannot stat ‘ww3_prnc.o’: No such file or directory
      Linking ww3_prnc
      *** file ww3_prnc.o not found ***
*** Missing object files ***
make: *** [/home/ww3/WW3/model/exe/ww3_prnc] Error 3

Please help!
Thanks in advance.

**Comment by JessicaMeixner-NOAA** [comment]
Usually when linking is an issue during compilation it's an issue with a library such as netcdf. The .inp files will not have an effect on your compilation.

**Comment by krishnanathira** [comment]
Thanks, Dr. Jessica. I have provided the paths for nc. config and nf. config before the w3_make step. Still showing the same error.

**Comment by aliabdolali** [comment, marked as answer]
could you try w3_make ww3_prnc instead of w3_automake?
if your netcdf is loaded, you can check
`nc-config --has-nc4`
if the answer is yes, then set the following:
```
  export WWATCH3_NETCDF=NC4
  export NETCDF_CONFIG=<path to nc-config>/nc-config
```
or
```
  setnev WWATCH3_NETCDF NC4
  setenv NETCDF_CONFIG <path to nc-config>/nc-config
```

**Reply by krishnanathira** [reply]
Thank you for your response.
I have installed nc-config and nf-config inside a folder **Libraries/grib2/bin**

```
> cd /home/ww3/WW3/Libraries/grib2/bin
> nc-config --has-nc4
> yes
```
after that, I did the following steps 

```
> cd /home/ww3/WW3/model/bin
> export WWATCH3_NETCDF=NC4
> export NETCDF_CONFIG=/home/ww3/WW3/Libraries/grib2/bin/nc-config
> w3_make ww3_prnc
```
sorry to say that, the same error persists.

**Reply by krishnanathira** [reply]
Hi Dr. Ali
I have rechecked the paths again, added one more line
```
export NETCDF=<path to netcdf>
export NETCDF_CONFIG=<path to nc-config>nc-config
export LD_LIBRARY_PATH=<path to library>
w3_make ww3_prnc
```
And it worked.
               ```
   *****************************
              ***   WAVEWATCH III make      ***
                *****************************
 
 Exporting WWATCH3_ENV=/home/ww3/WW3/model/bin/wwatch3.env
Main directory    : /home/ww3/WW3/model
Scratch directory : /home/ww3/WW3/model/tmp
Save source codes : yes
Save listings     : yes
Shared Object     : No
 
[INFO] list of the programs selected : ww3_prnc
```
Thanks a lot for your help.
